File: python/collatz-conjecture/collatz_conjecture.py
import itertools


def valid_number(f):
    def wrapper(number):
        if number < 1:
            raise ValueError("Only positive integers are allowed")
        return f(number)
    return wrapper


@valid_number
def steps(number):
    for step in itertools.count():
        if number == 1:
            break

        if number % 2 == 1:
            number = number * 3 + 1
        else:
            number /= 2

    return step

File: javascript/collatz-conjecture/collatz-conjecture.js
export const steps = (n) => {
    if (n <= 0)
        throw new Error('Only positive numbers are allowed'); 
         
    let i = 0;

    while (n != 1){
        if (n % 2)
            n = 3*n + 1
        else
            n /= 2
        i++
    }
    return i
};
